### DISCUSSION TITLE
Export, Animated ... Missing save option for single layer / alternatives

### DISCUSSION THREAD

**Discussion by hi5** [tannerhelland/PhotoDemon#513] [Ideas]
This may be a missing/not yet added feature so I didn't want to make it an issue as such.

I noticed that when I have a single layer image and choose Export, animated... you're shown a warning:

> This is a still image (only one frame of animation).
> You may proceed, but the image will be saved as a static image, not an animated one.

But as far as I can tell nothing is saved nor do you get the "Save as" dialog for the chosen format (gif, png etc)

I would expect one of these:

1. Simply show a warning you can't save a single layer image as animation and close it, basically what is happening now. This would make sense.
2. If you do want the user to allow Export as for single and multilayer images, actually implement the "Save as" chosen format for a single layer as indicated by the current warning text.
3. Perhaps adding Explicit `Export as PNG` (a Save As alias but while keeping the original format[1] e.g. PDI or what have you), `Export as Animated PNG` etc, OR combine the Export As formats into one dialog where you can tick a box for Animated / Single layer, so `Export as (Animated) PNG` -> dialog allows you choose between single layer/animation format.

I don't have a specific preference.

[1] If you do a `Save As` at the moment you continue with the file format you've saved it as, so you have a multi-layer PDI, save as PNG and continue to work on the image you might not remember to save it as a multi-layer format to avoid loosing any changes you've made. Hence an `Export As format` might make sense so you save it as as PNG while keeping the original format in your work space which may by a multi-layer format.

**Comment by tannerhelland** [comment]
All excellent suggestions, @hi5 .  Thank you so much for the thoughtful discussion.  (I also apologize for the broken "this isn't an animation so it will save as a single-frame image" behavior 🤦‍♂️ ugh)

I will fix the broken behavior first, then overhaul the Export menu a bit.  Originally there was only one animated export option (Export animated GIF) and as new animation formats have been added, breaking each format into its own menu makes less and less sense.

The new export menu should condense these into a single "Export animation" menu, alongside a new "Export image" or "Export as..." menu for single-frame images.  (Text TBD after I study Photoshop and other software's wording.)  

I will report back when I have something workable to share.  Thank you!

**Comment by tannerhelland** [comment]
Continuing my eternal trend of "wow that took longer than expected", the latest PhotoDemon nightly builds now include a much-improved `File > Export` menu.  There are new options for `Export image to file`, `Export layers to individual files`, and the old `Export to animated (GIF/JXL/PNG/WebP)` menus have been condensed to a single `Export animation` menu.  Single-frame animations *should* now be correctly identified and rerouted to the new `Export image` tool.

Thank you again for your thoughtful help on this topic!
